test/tests/selftest/pbs_syslog_tests.py

```diff
@@ -0,0 +1,146 @@
+# trademark licensing policies.
+
+from tests.selftest import *
+
+
+class TestSyslog(TestSelf):
+    """
+    Test pbs syslog logging
+    """
+
+    def setup_syslog(self, hostname=None, local_log=None,
+                     syslog_facility=None, syslog_severity=None):
+        """
+        Setup syslog in pbs.conf
+        """
+        a = {
+            'PBS_SYSLOG': syslog_facility,
+            'PBS_SYSLOGSEVR': syslog_severity,
+            'PBS_LOCALLOG': local_log}
+        self.du.set_pbs_config(hostname=hostname, confs=a, append=True)
+        PBSInitServices().restart()
+        self.assertTrue(self.server.isUp(), 'Failed to restart PBS Daemons')
+
+    def test_basic_syslog_match(self):
+        """
+        Basic test to check if logging via syslog works
+        Each individaul daemon is logged seperately
+        :return:
+        """
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=7)
+        a = {'Resource_List.ncpus': 1}
+        J1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        msg_str = "Job;" + jid1 + ";Job Run at request of Scheduler"
+        self.server.log_match(msg_str, n=500)
+
+    def test_basic_syslog_severity(self):
+        """
+        Test that that PBS logs messages via syslog
+        according to the set priority
+        :return:
+        """
+        self.t = int(time.time())
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=6)
+        a = {'Resource_List.ncpus': 1}
+        J1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        msg_str = "Job;" + jid1 + ";Considering job to run"
+        with self.assertRaises(PtlLogMatchError):
+            self.scheduler.log_match(
+                msg_str, n=500, max_attempts=10, starttime=self.t)
+        self.server.deljob(jid1, wait=True)
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=7)
+        J2 = Job(TEST_USER, attrs=a)
+        jid2 = self.server.submit(J2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        msg_str = "Job;" + jid2 + ";Considering job to run"
+        self.scheduler.log_match(msg_str, n=500, max_attempts=10)
+
+    def test_muti_host_syslog_match(self):
+        """
+        Test that syslog matching works on a multihost system
+        :return:
+        """
+        self.setup_syslog(local_log=0, syslog_facility=9, syslog_severity=7)
+        if len(self.moms) != 2:
+            self.skip_test(reason="need 2 mom hosts: -p moms=<m1>:<m2>")
+
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr = {'Resource_List.select': '2:ncpus=1'}
+        j = Job(TEST_USER, attrs=attr)
```

Comment:
nit: in the previous tests you used capitals for jobs.  Here you used a little j.  Consider capitalizing it.

Reply:
Changed these test steps